frontend/lib/create_listing/view/create_listing.view.dart

```diff
@@ -0,0 +1,414 @@
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:provider/provider.dart';
+import 'package:utm_marketplace/create_listing/model/create_listing.model.dart';
+import 'package:utm_marketplace/create_listing/view_models/create_listing.viewmodel.dart';
+import 'package:utm_marketplace/shared/components/loading.component.dart';
+
+class CreateListingView extends StatefulWidget {
+  const CreateListingView({super.key});
+
+  @override
+  State<CreateListingView> createState() => _CreateListingViewState();
+}
+
+class _CreateListingViewState extends State<CreateListingView> {
+  final _formKey = GlobalKey<FormState>();
+  late CreateListingViewModel viewModel;
+  final _titleController = TextEditingController();
+  final _priceController = TextEditingController();
+  final _descriptionController = TextEditingController();
+  bool _showValidationErrors = false;
+
+  @override
+  void initState() {
+    super.initState();
+    viewModel = Provider.of<CreateListingViewModel>(context, listen: false);
+  }
+
+  // Header widgets
+  final appBarTitle = const Text(
+    'Create Post',
+    style: TextStyle(
+      fontSize: 32,
+      fontWeight: FontWeight.bold,
+    ),
+  );
+
+  final postButton = const Text(
+    'Post',
+    style: TextStyle(
+      color: Color(0xFF2F9CCF),
+      fontSize: 16,
+    ),
+  );
+
+  // Form widgets
+  Widget _buildImageSelector() {
```

Comment:
Let's make a separate component for this. This page is quite complicated so we should probably define entire component widgets rather than giant helpers like this.